src/plugins/files/server/file_service/internal_file_service.ts

```diff
@@ -64,6 +69,14 @@ export class InternalFileService {
     await file.delete();
   }
 
+  public async bulkDeleteFiles({
+    ids,
+  }: BulkDeleteFilesArgs): Promise<Array<PromiseSettledResult<void>>> {
+    const promises = ids.map((id) => bulkDeleteConcurrency(() => this.deleteFile({ id })));
```

Comment:
I know there's complexities here that I'm not aware of but would it be possible to use the saved object client's bulkDelete functionality?

Reply:
If you look at the private `get` below you can see that there is some logic encapsulated. It converts the id to a `File` instance and calls its `delete()` method which then delete the content from the blob storage, calls the `updateFileState` to change the state, delete the SO and finally logs some events `logAuditEvent` 😊 

That's what we hook into.

Reply:
+1 to what @sebelga mentioned, essentially, before event deleting the saved object metadata document, we need to delete the blob contents first, which is stored in a different place.